R/analyse_pseudo_bulk.R

```diff
@@ -0,0 +1,331 @@
+    
+    result <- split_clustering(seurat_object, group_by,resolution_, 1, cluster1_,cluster2_)
+    return(result)
+  }
+  
+  if(split_by == "Louvain_multilevel"){
+    resolution_ <- k_variable[[1]]
+    cluster1_ <- k_variable[[2]]
+    cluster2_ <- k_variable[[3]]
+   
+    result <- split_clustering(seurat_object, group_by,resolution_, 2, cluster1_,cluster2_)
+    return(result)
+  }
+  
+  if(split_by =="SLM"){
+    resolution_ <- k_variable[[1]]
+    cluster1_ <- k_variable[[2]]
+    cluster2_ <- k_variable[[3]]
+    
+    result <- split_clustering(seurat_object, group_by,resolution_, 3, cluster1_,cluster2_)
+    return(result)
+  }
+  
+  if(split_by =="Leiden"){
+    resolution_ <- k_variable[[1]]
+    cluster1_ <- k_variable[[2]]
+    cluster2_ <- k_variable[[3]]
+    
+    result <- split_clustering(seurat_object, group_by,resolution_, 4, cluster1_,cluster2_)
+    return(result)
+  }
+})
+
+
+
+#' generate_pseudo_bulk_data using SingleCellExperiment
+#' @inherit generate_pseudo_bulk_data 
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param split_by      variable -> split by a variable within the metadata; k must be a string
+#'                      random -> splits based on a random number; k must be a number
+#'                      subclustering [resolution, cluster_1, cluster_2]
+#' @param k_variable    variable dependent on the split_by
+#'
+#' @returns             returns pseudo bulk generated data
+#' @examples
+#'
+#' @export
+setMethod("generate_pseudo_bulk_data", c("object" = "SingleCellExperiment"), function(object, 
+                                                                                      group_by, 
+                                                                                      split_by, 
+                                                                                      k_variable){
+  if (split_by == "variable"){
+    result <- split_variable_sce(object, group_by, k_variable)
+    return(result)
+  }
+  
+  if (split_by == "random"){
+    result <- split_random_sce(object, group_by, k_variable)
+    return(result)
+  }
+  
+  if(split_by == "subclustering"){ 
+    result <- split_subclustering_sce(object, group_by, k_variable)
+    return(result)
+  }
+    
+})
+
+
+
+
+#' split SCE Object by variable
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param k_variable    variable for sub setting must be in the metadata
+#' 
+#' @returns             returns pseudo bulk generated data
+#' @importFrom SingleCellExperiment scuttle 
+#' 
+#' @examples
+#' # SCE_RESULT_VARIABLE <- split_variable_sce(SCE_OBJECT, GROUP_BY,K_VARIABLE)
+#' 
+split_variable_sce <- function(sce_object, group_by, k_variable){
+  
+  aggregated_object <- scuttle::aggregateAcrossCells(sce_object, ids=SingleCellExperiment::colData(sce_object)[,c(group_by, k_variable)])
```

Comment:
I'd suggest to add an explicit check to test that the metadata field exists (with a nice error message)

Reply:
This should still be addressed